Sources/Core/Shared/Operation.swift

```diff
@@ -79,27 +79,24 @@ public class Operation: NSOperation {
         }
     }
 
-    class func keyPathsForValuesAffectingIsExecuting() -> Set<NSObject> {
-        return ["State"]
-    }
-
-    class func keyPathsForValuesAffectingIsFinished() -> Set<NSObject> {
-        return ["State"]
-    }
-
-    class func keyPathsForValuesAffectingIsCancelled() -> Set<NSObject> {
-        return ["Cancelled"]
+    private enum NSOperationKeyPaths: String {
+        case Cancelled = "isCancelled"
+        case Executing = "isExecuting"
+        case Finished = "isFinished"
     }
```

Comment:
Just thinking a little bit down the road, with Swift 3.0. It would probably be better to add this to `Foundation.NSOperation` instead. Because soon "NSOperation" is not going to be a thing.

```swift
private extension Foundation.NSOperation {
    enum KeyPaths: String {
        case Cancelled = "isCancelled"
        case Executing = "isExecuting"
        case Finished = "isFinished"
    }
}
```